### PR TITLE
Make VR validation of null more consistent

### PR DESCRIPTION
**Pull request by mrbean-bremen**
- DicomElement.ValueRepresentation.ValidateString(null) now always throws DicomValidationException
- see #1590

#### Checklist
- [x] The pull request branch is in sync with latest commit on the *fo-dicom/development* branch
- [ ] I have updated API documentation - n/a
- [x] I have included unit tests
- [x] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file

### CODE REVIEW

File: FO-DICOM.Core/DicomElement.cs [hunk 1]
```diff
@@ -199,10 +199,7 @@ private byte[] StringToBytes()
             return bytes;
         }
 
-        protected override void ValidateString()
-        {
-            ValueRepresentation?.ValidateString(_value);
-        }
+        protected override void ValidateString() => ValueRepresentation?.ValidateString(StringValue);
```
Comment (mrbean-bremen) [inline FO-DICOM.Core/DicomElement.cs:202]: This somewhat changes the behavior: before, if the string value has not been decoded yet, the validation was either skipped (for the VRs where `string.IsNullOrEmpty` was used), or would cause a `NullReferenceException`. I don't think that it has an impact on the performance in practice, as otherwise we would have seen these exceptions all over the place. Also the benchmark results did not change.

Comment (gofal) [reply FO-DICOM.Core/DicomElement.cs:202]: Validation is triggered if some content is added to a DicomDataset via methods "AddOrUpdate" or "Add". In that case there is always a decoed string passed.
Only if dicomdatasets are parsed from a network or filestream, then (for performance reasons) the raw binary stream is kept im memory and only decoded when the StringValue is first accessed (lazy decoding). An in these cases because of the mantra to be as tollernat as possible when receiving data, there is no validation.

So basically it should never happen, that validation is called for a tag, where the string value is still in a not-decoded-state. Did you observe that?

**Review by gofal — APPROVED** [review]

File: FO-DICOM.Core/DicomElement.cs (hunk 1 above)
Comment (mrbean-bremen) [reply FO-DICOM.Core/DicomElement.cs:202]: No, I did not observe that, and you are right - this will not happen, it is only a theoretical possibility.
